CMakeLists.txt

```diff
@@ -1,59 +1,119 @@
 # Require at least version 2.8.12 of CMake
 cmake_minimum_required(VERSION 2.8.12)
 
+if(CMAKE_VERSION VERSION_EQUAL 3.0 OR CMAKE_VERSION VERSION_GREATER 3.0)
+    cmake_policy(SET CMP0042 NEW)
```

Comment:
A short comment here to explain what this does and why it is necessary would be nice.